modules/slugs/includes/class-qtranslate-slug.php

```diff
@@ -1476,9 +1476,10 @@ public function qts_modify_term_form( $term ) {
         echo "<script type=\"text/javascript\">\n// <![CDATA[\r\n";
         echo "
 		var slugforms = jQuery('#qts_term_slugs').html();
-		jQuery('#slug').parent().html(slugforms)\n;
-		console.log(slugforms);
-		
+                if ( jQuery('#slug').length ) {
+                    jQuery('#slug').parent().html(slugforms);
+                    jQuery('#qts_term_slugs').remove();
```

Comment:
Why is that `qts_term_slugs` even created in first hand? In which cases do we want this?

Reply:
Why is that `slugforms` added in Javascript? Can't we simply keep `#qts_term_slugs` and update the fields on language change? That was a hack but it looks really weird.

Reply:
`qts_term_slugs` is created as default input for taxonomies edit pages (e.g. for `edit-tags.php?taxonomy=category`), and original slug field is hidden.
For single term pages the `qts_term_slugs` does not fit well with the existing page layout, hence code is copied in place of the original slug input (if exists) placed in the existing table and `qts_term_slugs` is removed.
I guess the intent is just to reuse the same actions for both cases above.
As there is no dedicated hooks here (afaik) to make things cleaner, any possible approach would be hacky. Hence I retained the original philosophy...

Reply:
OK I see. Perhaps there's a better way, as there are two different hooks (add/edit), but let's keep it like this for now.

However there's another part I don't understand, not coming from this patch but maybe you know:
https://github.com/qtranslate/qtranslate-xt/blob/dd241f42b6195049273c844d23f4d6d81899b695/modules/slugs/includes/class-qtranslate-slug.php#L1290-L1291

I can't find when this is triggered. When we edit a term it goes in the `else` part.

Reply:
Ok, I think that that condition intention was the same currently accomplished with javascript.
This is not working (maybe with some old WP release it was), hence it is always evaluated `false `and maybe later the jquery patch was introduced and this has been left behind as junk code...
Initial intention was just to fix current situation, but actually this needs to be fully revised: I've added fe5961f2b504d70434e99b471e1f6ef350d5c273

- `qts_modify_term_form` removed
- callbacks for `_add_form_fields `and `_add_form_fields `are differentiated, no need for conditions
- only hack left is `hide_term_slug_box `which hides through jquery the original slugs fields, but there is no other way afaik
- removed apparently orphan function `qts_insert_term_input `

Reply:
Great, this looks better! I have other comments but for that part we should be fine.